spec/core/query_spec.rb

```diff
@@ -334,4 +334,14 @@
       end
     end
   end
+  describe "apply_where" do
+    context Daru::Vector do
+      it "matches regexp with a block input" do
+        dv = Daru::Vector.new ['3 days', '5 weeks', '2 weeks']
+        dv = dv.apply_where(dv.match /weeks/) { |x| "#{x.split.first.to_i * 7} days" }
+        expected_dv = Daru::Vector.new ['3 days', '35 days', '14 days']
+        expect(dv).to eq expected_dv
+      end
+    end
```

Comment:
a) modern tests please (with `subject` and one-statement checks, see betterspecs.org)
b) you need to check several conditions, like "empty boolean vector", "block returning something unexpected" and so on.